free-1.md

```diff
@@ -11,41 +11,100 @@ _What need is met by your idea? Who are you helping? What is that person's
+what progress you have made in the project while following the pattern. You
+could also add your own notes, or edit the pattern yourself using the language.
+
 ### Why you?
 
 _What excites you about this idea? How did you come up with it?_
 
+I like to crochet, and I find it difficult to keep track of where I am in my
+project when I am either following a pattern, partially following a pattern
+while making my own changes, or when I have created my own projects without a
+pattern. I came up with this idea because I would like there to be an easy way
+to create crochet patterns, and I am already knowledgeable about this domain. It
+is also a cool domain because it already has it's own abbreviations (it's own
+language) that could be used in the language.
+
 ### Domain
 
 _Describe the project's domain in five words._
 
+Crochet pattern creation and use.
+
 ### Interface (syntax)
 
 _How might the user interact with the language? What does programming look
 like? Why is this the right way to interact with the problem domain?_
 
+I imagine three different ways to interact with it. First, there would be the
+coding that a user could do to create a pattern. Then, this coding could be
+exported to either a PDF, or to an interactive program where the user gets to
+keep track of what instruction they are on. I think the coding is the right
+domain because it would allow the user to easily edit and create a pattern.
+They could also add stylistic choices for the PDF output. The PDF output is
+appropriate because that is how crochet patterns are currently circulated. And
+the interactive program that you load the code into would be very helpful for
+people working on crochet projects.
+
 ### Operation (semantics)
 
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
 
+When the program runs, either a PDF or the interactive program would be created.
+
+The program interacts with the user through the interactive system, where the
+user can update what instruction they are on and add extra notes for each
+instruction. They can also maybe scale the project with crochet hook and yarn
+size.
+
+An error that might occur would be if the number of stitches does not line up
+on a project. This would be a runtime error, and would help the pattern
+creator make their pattern functional.
+
 ### Expressiveness
 
 _What should be easy to do in this language? What should be possible, but
 difficult? What should be impossible or very difficult?_
 
+It would be easy to create crochet patterns with the variety of basic crochet
+rules. It would be possible, but more difficult to define your own crochet
+step that users could create. Maybe this could link to a youtube video with a
+further description of how the step worked. It should be impossible or very
+difficult to create something outside of the domain of crochet pattern making.
+
```

Comment:
Would it be possible to upload patterns from crocheting-adjacent activities? Weaving, knitting, etc. I suspect that this may be possible (but slightly difficult) depending on how you define your DSL. Then, the question is do you *want* to include this in your scope, or focus the domain more?